src/Umbraco.Web.Common/Extensions/WebApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,36 @@
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Hosting;
+using Umbraco.Cms.Core.Configuration.Models;
+using Umbraco.Cms.Core.DependencyInjection;
+
+namespace Umbraco.Extensions;
+
+/// <summary>
+/// Extension methods for <see cref="WebApplicationBuilder" />.
+/// </summary>
+public static class WebApplicationBuilderExtensions
+{
+    /// <summary>
+    /// Creates an <see cref="IUmbracoBuilder" /> and registers basic Umbraco services.
+    /// </summary>
+    /// <param name="builder">The builder.</param>
+    /// <returns>
+    /// The Umbraco builder.
+    /// </returns>
+    public static IUmbracoBuilder CreateUmbracoBuilder(this WebApplicationBuilder builder)
+    {
+        // Configure Umbraco defaults, but ignore decorated host builder and
+        // don't add runtime as hosted service (this is replaced by the explicit BootUmbracoAsync)
+        builder.Host.ConfigureUmbracoDefaults(false);
+
+        // Do not enable static web assets on production environments,
+        // because the files are already copied to the publish output folder.
+        if (builder.Configuration.GetRuntimeMode() != RuntimeMode.Production)
+        {
+            builder.WebHost.UseStaticWebAssets();
+        }
```

Comment:
Although I don't expect a lot of users will be running a non-Development environment using the Production runtime mode from source (non-published output), but if they do, the files from RCLs won't be available and return 404s.

This can be easily fixed by manually calling `builder.WebHost.UseStaticWebAssets();` yourself. We could add this to the documentation, but I guess it's really an edge-case and could potentially result in users adding this without knowing/fully understanding the implications and not removing/disabling it on production again...